ckanext/oht/plugin.py

```diff
@@ -59,6 +65,38 @@ def before_update(self, context, current, resource):
             _update_resource_last_modified_date(resource, current=current)
         return resource
 
+    def get_dataset_labels(self, dataset_obj):
+        """
+        Stops private datasets from being visible to other members of the
+        same organisation. Only the creator and collaborators can see them.
+        """
+        labels = super(OHTPlugin, self).get_dataset_labels(dataset_obj)
+
+        if dataset_obj.owner_org:
+            labels.remove(u'member-%s' % dataset_obj.owner_org)
+            labels.append(u'creator-%s' % dataset_obj.creator_user_id)
```

Comment:
This is the key code, overriding the behaviour defined here:
https://github.com/ckan/ckan/blob/e10534178b58a5f3ac737505aa2a31450b09ad2e/ckan/lib/plugins.py#L656-L659